File: Backjoon/2212_센서.swift
/*
 센서
 https://www.acmicpc.net/problem/2212
*/

let n = Int(readLine()!)!
let k = Int(readLine()!)!
var sensor = readLine()!.split(separator: " ").map { Int("\($0)")! }
if n <= k {
    print(0)
} else {
    sensor.sort()
    var gap: [Int] = []
    for i in 1..<sensor.count {
        gap.append(sensor[i] - sensor[i-1])
    }
    gap.sort()
    gap.removeLast(k-1)
    print(gap.reduce(0,+))
}

File: Backjoon/Python/2212_센서.py
"""
 센서
 https://www.acmicpc.net/problem/2212
"""

n = int(input())
k = int(input())
sensor = list(map(int,input().split()))
if n <= k:
    print(0)
else:
    sensor.sort()
    gap = []
    for i in range(1,len(sensor)):
        gap.append(sensor[i] - sensor[i-1])
    gap.sort()
    gap = gap[:len(gap) - (k - 1)]
    print(sum(gap))
